syntax = "proto3";

service UserService {
    rpc Register (RegisterRequest) returns (RegisterReply) {}
    rpc Authenticate (AuthenticateRequest) returns (AuthenticateReply) {}
    rpc UserGetInformation (UserGetInformationRequest) returns (UserGetInformationReply) {}
}

message RegisterRequest {
    string user_name = 1;
    string password = 2;
}

message RegisterReply {
    int32 status = 1;
}

message AuthenticateRequest {
    string user_name = 1;
    string password = 2;
}

message AuthenticateReply {
    int32 status = 1;
}

message UserGetInformationRequest {
    int32 user_id = 1;
}

message UserGetInformationReply {
    int32 status = 1;
    string user_name = 2;
}
